xla/service/cpu/onednn_matmul.cc

```diff
@@ -177,6 +278,93 @@ ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_OneDnnMatMul(
   matmul_prim.execute(onednn_stream, matmul_args);
 }
 
+ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_OneDnnMatMulReorder(
+    void* result, void** args) {
+  // args[0]: ptr to nargs
+  // args[1]: ptr to ExecutableRunOptions
+  // args[2]: ptr to OneDnnMatMulConfig
+  // args[3...]: ptrs to operands
+  int arg_indx = 0;
+  const int64_t num_args = *(static_cast<int64_t*>(args[arg_indx++]));
+
+  const xla::ExecutableRunOptions* run_options =
+      static_cast<const xla::ExecutableRunOptions*>(args[arg_indx++]);
+
+  engine cpu_engine(engine::kind::cpu, 0);
+
+#ifndef ENABLE_ONEDNN_OPENMP
+  tsl::OneDnnThreadPool thread_pool;
+  auto onednn_stream = [&] {
+    if (run_options != nullptr &&
+        run_options->intra_op_thread_pool() != nullptr) {
+      thread_pool = tsl::OneDnnThreadPool(
+          run_options->intra_op_thread_pool()->getPool(), false);
+      return stream(
+          dnnl::threadpool_interop::make_stream(cpu_engine, &thread_pool));
+    } else {
+      return stream(cpu_engine);
+    }
+  }();
+#else
+  auto onednn_stream = stream(cpu_engine);
+#endif  // ENABLE_ONEDNN_OPENMP
```

Comment:
This code is similar to what `__xla_cpu_runtime_OneDnnMatMul` uses. Please refactor it into a function.